### PR TITLE
[MI-2082] Add area path field in create task modal

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/types/common/form.d.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 type FieldType = 'dropdown' | 'text' | 'timestamp'
 type ValidationTypes = 'isRequired' | 'maxCharLen' | 'minCharLen' | 'regex' | 'regexErrorMessage'
 type SubscriptionModalFields = 'organization' | 'project' | 'eventType' | 'channelID' | 'timestamp'
-type CreateTaskModalFields = 'organization' | 'project' | 'type' | 'title' | 'description' | 'timestamp'
+type CreateTaskModalFields = 'organization' | 'project' | 'type' | 'title' | 'description' | 'areaPath' | 'timestamp'
 type ErrorComponents = 'SubscribeModal' | 'CreateTaskModal'
```
Comment (muazzamqazi) [inline webapp/src/types/common/form.d.ts:5]:
```suggestion
type CreateTaskModalFields = 'organization' | 'project' | 'type' | 'title' | 'description' | 'areaPath' | 'timestamp';
type ErrorComponents = 'SubscribeModal' | 'CreateTaskModal';
```

**Review by muazzamqazi — APPROVED** [review]

File: plugin.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
                 "key": "EncryptionSecret",
                 "display_name": "Encryption Secret:",
                 "type": "generated",
-                "help_text": "The secret key used to encrypt and decrpyt OAuth token."
+                "help_text": "The secret key is used to encrypt and decrypt the OAuth token.\nRegenerating the secret key will cause the loss of data."
```
Comment (avas27JTG) [inline plugin.json:46]:
```suggestion
                "help_text": "The secret key used to encrypt and decrypt the OAuth token.\nRegenerating the secret key will cause the loss of data."
```

is this text verified and updated as per Mike suggestion? if not pls check with him and get it updated

File: server/plugin/utils.go [hunk 3]
```diff
@@ -31,17 +31,28 @@ func (p *Plugin) sendEphemeralPostForCommand(args *model.CommandArgs, text strin
 }
 
 // DM posts a simple Direct Message to the specified user
-func (p *Plugin) DM(mattermostUserID, format string, args ...interface{}) (string, error) {
+func (p *Plugin) DM(mattermostUserID, format string, isSlackAttachment bool, args ...interface{}) (string, error) {
```
Comment (avas27JTG) [inline server/plugin/utils.go:34]: sync with master and see if there is a need to make changes for wherever `DM` is used

Comment (ayusht2810) [reply server/plugin/utils.go:34]: I already did the changes wherever `DM` was present while adding this flag.

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**